app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php

```diff
@@ -412,14 +412,22 @@ private function getCountFromCategoryTableBulk(
                 []
             )
             ->where('ce.entity_id IN (?)', $categoryIds);
-
         $connection->query(
             $connection->insertFromSelect(
                 $selectDescendants,
                 $tempTableName,
                 ['category_id', 'descendant_id']
             )
         );
+        foreach ($categoryIds as $catId) {
+            $connection->insert(
```

Comment:
Calling SQL queries in loop might have a negative performance impact.
Please use $connection->insertMultiple instead.